src/QsCompiler/LanguageServer/EditorState.cs

```diff
@@ -153,6 +153,33 @@ internal bool QsProjectLoader(Uri projectFile, out ProjectInformation info)
             return true;
         }
 
+        internal bool QsTemporaryProjectLoader(Uri sourceFileUri, string sdkVersion, out Uri projectUri, out ProjectInformation info)
+        {
+            projectUri = null;
+            info = null;
+
+            var localFolderPath = Path.GetDirectoryName(sourceFileUri.LocalPath);
+            var projectFolderPath = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), "qsharp", Guid.NewGuid().ToString())).FullName;
+            var projectFilePath = Path.Combine(projectFolderPath, $"generated.csproj");
+            using (var outputFile = new StreamWriter(projectFilePath))
+            {
+                outputFile.WriteLine(
+                    TemporaryProject.GetFileContents(
+                        compilationScope: Path.Combine(localFolderPath, "*.qs"),
```

Comment:
Yes, I think your understanding is correct. I see now why this is a problem, especially for those cases where both projects remain alive and are trying to compile the same .qs files. Thanks for pointing this out.

Reply:
It looks like if we generate a stable filename for the temporary .csproj (e.g. by hashing the folder path instead of creating a GUID), then the same .csproj is used every time across create/delete/rename, and we avoid these problems. I think that fixes the issues you're describing, but please let me know if you have concerns with that approach.

Reply:
I think that should be fine - please try out how the user experience is in practice for python as a driver when you have multiple folders with Q# files that you want to use.

Reply:
Actually, I think the case where you may run into trouble there is when you have multiple folders with .qs files. Having these files all open in the IDE, and editing them I think would cause there to be a change notification for a file that is not tracked anywhere - it might actually throw. So that would need to be handled somehow. Maybe you make room for having multiple such auto-generated csproj, one for each subfolder. I haven't though about it in more detail, but let me know if you want to bounce ideas.

Reply:
Sorry, I keep coming back to this: You also run into trouble when you copy over a .qs file without opening it in the IDE - then technically our auto-generated project changed, but that is never registered because the autogenerated project file is not listed among the project files in the workspace.

Reply:
Thanks Bettina, that all makes sense, I appreciate the help thinking through the edge cases here!

For the multiple-folder case, this appears to be already working, because the temporary project is placed in a folder that is based on the hash of the folder containing the `.qs` files. So each folder of `.qs` files gets a unique temporary project, and they don't conflict.

For the case where `.qs` files are added/modified/deleted outside the IDE, you're absolutely right, I wasn't picking up those changes. I've now tried to make the minimal change to do what you've suggested: when creating the temporary project, add a `FileSystemWatcher` for `*.csproj` files in that folder. It seems to work, from what I can tell - the events are fired in the code and compilations are updated even when changing the files from outside the IDE.

Reply:
Lgtm, though that is definitively something that can profit from a couple of people actually using it to see if there is are any more things we haven't thought of. Off the top of my mind nothing further comes to mind, so I am looking forward to seeing this in action. :)